### ISSUE TITLE
Why rmse_val and rmse_trn differ when --restart the training?

### ISSUE THREAD

**Issue by njzjz** [deepmodeling/deepmd-kit#1698]
### Discussed in https://github.com/deepmodeling/deepmd-kit/discussions/1696

<div type='discussions-op-text'>

<sup>Originally posted by **Lewis-YL** May  9, 2022</sup>
Hello everyone,

Why rmse_val and rmse_trn are different when using --restart or --init-model?

In my training output file "lcurve.out"
At step 1000000:
rmse_val: 4.55e-01    
rmse_trn: 2.36e-01

If using the command:
dp train input.json --restart model.ckpt-1000000
At step 100000:
rmse_val: 2.92e+00
rmse_trn: 2.41e+00

If using the command:
dp train input.json --init-model model.ckpt-1000000
At  step 0:
rmse_val: 9.19e+00
rmse_trn: 8.06e+00

</div>

**Comment by njzjz** [comment]
The random state is not recovered, so they visit the different data to test.

**Comment by njzjz** [comment]
It looks meaningless to recover the random state, as the state of `DeepmdData` is not recovered.

**Comment by wanghan-iapcm** [comment]
`restart` and `init-model` are different things. Use `restart` to continue the training from a check point. Please read
https://github.com/deepmodeling/deepmd-kit/blob/master/doc/train/training-advanced.md#options-and-environment-variables

**Comment by Lewis-YL** [comment]
The tutorial just says init-model initializes the model at checkpoint.

Can you tell some details about 'initialize'?

Does 'initialize' not only reset the index number, but also erase the pretrained feature of NN?
If so, what is the difference if using dp train input.json without --init-model?

**Comment by Lewis-YL** [comment]
> `restart` and `init-model` are different things. Use `restart` to continue the training from a check point. Please read https://github.com/deepmodeling/deepmd-kit/blob/master/doc/train/training-advanced.md#options-and-environment-variables

The tutorial just says init-model initializes the model at checkpoint.

Can you tell some details about 'initialize'?

Does 'initialize' not only reset the index number, but also erase the pretrained feature of NN?
If so, what is the difference if using dp train input.json without --init-model?

**Comment by njzjz** [comment]
Do you compare only the total loss, or energy loss and force loss?

The total loss, specifically the pre-factor, is a function of the step index.

**Comment by Lewis-YL** [comment]
I see!

So pre-trained feature

> Do you compare only the total loss, or energy loss and force loss?
> 
> The total loss, specifically the pre-factor, is a function of the step index.

Problem solved!

So, the pre-trained feature is stilled stored in NN when using --restart.
The larger loss due to reset of pre-factor.
So we need to adjust prefactor during --init-model in order to get less tot error.

Thank you!

**Comment by njzjz** [comment]
See eq. 10 in https://tutorials.deepmodeling.com/en/latest/Tutorials/DeePMD-kit/learnDoc/Theory.html.
